### PR TITLE
Feature Rachel iteration3

### PR DESCRIPTION
**Pull request by rwilliams659**
---

project: ROMCOM 
subject: PULL REQUEST TEMPLATE
program: TURING MOD1 FE (paired project)
title: GENERAL PROJECT PULL REQUEST TEMPLATE
participants: Rachel Williams and Suzanne Wolf
project manager: Leta

---

## Description of Changes 

**Iteration 3 is now complete:**

- collectUserInput function was created to save user input from Make Your Own Cover form into existing arrays for cover images, titles, and descriptors
- Extraneous comma was deleted from titles array to prevent 'undefined' from displaying when a random title is displayed
- createNewCover function was created to generate Book class instance
- getRandomBookCover was refactored to call createNewCover function
- Make Your Own Cover button was hidden from form view

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [x] Refactoring

# How Has This Been Tested?
Within each newly created function, each time we mutated an array, updated a variable value, or altered a class, we checked that the site was performing as expected and that all values had also been updated as expected. We then checked functions as a whole in the same way. 

# CHECKLIST:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by GreenbergKU** [comment]
I was playing around with the code and noticed that if the form is not filled out, but you press 'make my book', the separate arrays (covers, titles, descriptors) each push a value of "", which will cause problems for the findRandomBookCover().

File: src/data.js [hunk 1]
```diff
@@ -24,7 +25,7 @@ var titles = [
   "Stolen Hearts",
   "Secrets and Silk",
   "Dreams of Fire",
-  "Lovers and Enemies",,
```
Comment (GreenbergKU) [inline src/data.js:27]: Great find!  Explains why the covers.length was constantly returning numbers inconsistent with what we believed it should be!

**Comment by rwilliams659** [comment]
Great catch! Let's add it to our task list tomorrow to make sure that the form inputs are not empty strings before we push those values into their respective arrays.

File: src/main.js [hunk 2]
```diff
@@ -52,33 +60,52 @@ function displayCurrentCover() {
...
-}
\ No newline at end of file
+  makeYourOwnButton.classList.remove("hidden");
+}
+
+function collectUserInput() {
+  event.preventDefault();
+  covers.push(userImageInput.value);
+  titles.push(userTitleInput.value);
+  descriptors.push(userDescriptor1.value, userDescriptor2.value);
```
Comment (GreenbergKU) [inline src/main.js:102]: this is what my PR comment is referring to... we still met Iteration 3 requirements but should fix this.

File: src/main.js [hunk 3]
```diff
@@ -52,33 +60,52 @@ function displayCurrentCover() {
...
-  randomCoverButton.classList.add("hidden"); 
+  savedViewSection.classList.add("hidden");
+  formViewSection.classList.remove("hidden");
+  randomCoverButton.classList.add("hidden");
   saveCoverButton.classList.add("hidden");
   homeButton.classList.remove("hidden");
 //*************ADDED code BELOW to ITER2 ********
   viewSavedButton.classList.remove("hidden");
+  makeYourOwnButton.classList.add("hidden");
+  //hide makeyourownbutton
```
Comment (GreenbergKU) [inline src/main.js:71]: Despite this not being in the directions for the project, it just makes logical sense!  Why have a button to navigate you to the page you are currently on?  I agree with this addition!

File: src/main.js [hunk 4]
```diff
@@ -52,33 +60,52 @@ function displayCurrentCover() {
...
-  formViewSection.classList.add("hidden");  
-  randomCoverButton.classList.add("hidden"); 
+  formViewSection.classList.add("hidden");
+  randomCoverButton.classList.add("hidden");
   saveCoverButton.classList.add("hidden");
   homeButton.classList.remove("hidden");
 //*************ADDED code BELOW to ITER2 ********
-  viewSavedButton.classList.add("hidden"); 
+  viewSavedButton.classList.add("hidden");
+  makeYourOwnButton.classList.remove("hidden");
```
Comment (GreenbergKU) [inline src/main.js:83]: I concur, if we hide it while in form view, guess we have to show it when in Saved View!

File: src/main.js [hunk 5]
```diff
@@ -52,33 +60,52 @@ function displayCurrentCover() {
...
+  formViewSection.classList.add("hidden");
+  savedViewSection.classList.add("hidden");
+  homeButton.classList.add("hidden");
+  randomCoverButton.classList.remove("hidden");
   saveCoverButton.classList.remove("hidden");
 //*************ADDED code BELOW to ITER2 ********
   viewSavedButton.classList.remove("hidden");
-}
\ No newline at end of file
+  makeYourOwnButton.classList.remove("hidden");
```
Comment (GreenbergKU) [inline src/main.js:95]: logic agrees, show the button!

**Comment by GreenbergKU** [comment]
We still have completed all the steps to iteration 3 so this problem shouldn't prevent us from officially considering Iteration 3 COMPLETE!  But I agree with you, we should definitely address this issue as our first step of Iteration 4 tomorrow!  Good work today!  Your PRE-WORK is so helpful, (your pseudo-code and code ideas shows me where you're at) so I appreciate your effort!

**Pull request merged**